cli/tools/coverage/mod.rs

```diff
@@ -347,7 +347,12 @@ fn create_reporter(
 }
 
 trait CoverageReporter {
-  fn report(&mut self, coverage_report: &CoverageReport, file_text: &str);
+  fn report(
+    &mut self,
+    coverage_report: &CoverageReport,
+    file_text: &str,
+    out_writer: &mut Box<dyn Write>,
```

Comment:
@bartlomieju I don't think its good idea to add another parameter to report
() ? because for `PrettyCoverageReporter` this arg is unnecessary

Reply:
Agree, pass this parameter to `LcovCoverageReporter` directly

Reply:
@bartlomieju I have changed the approach a bit  !!